### ISSUE TITLE
Add remarks about attribute ordering or x:Key attribute positioning when using resource dictionaries from other files

### ISSUE THREAD

**Issue by butler1233** [dotnet/docs-desktop#162]
Hi,

I've just been caught out by something quite odd, which doesn't appear to be covered in the docs. The following StackOverflow link told me the solution, and I think it should be documented. This is at least the case in .Net Framework 4.7.2, although it may have been fixed in .net core versions of WPF.
 
[The StackOverflow question that helped me out](https://stackoverflow.com/questions/15553446/cannot-find-resource-named-x-why-not)

### The issue

When referencing resources in an external file, the application will fail at runtime with an error as follows:

**System.Windows.Markup.XamlParseException:** ''Provide value on 'System.Windows.StaticResourceExtension' threw an exception.' Line number 'x' and line position 'y'.'

Inner Exception:

Exception: Cannot find resource named '{resourceName}'. Resource names are case sensitive.

### A Workaround

For some reason in this case the XAML parser is either considering the attribute order for the resources or only checking the first attribute, but either way, a resource defined like this will fail:

```xaml
<DataTemplate DataType="newMySqlModels:ThirdPartyLogisticsOwnerDefinition" x:Key="TplOwnerTemplate"/>
```

However, if the `x:Key` attribute is moved to be the first attribute like this, it works absolutely fine.

```xaml
<DataTemplate x:Key="TplOwnerTemplate" DataType="newMySqlModels:ThirdPartyLogisticsOwnerDefinition"/>
```

In both cases, the dictionary which contains the resource is referenced like this:

<UserControl.Resources>
        <ResourceDictionary>
            <ResourceDictionary.MergedDictionaries>
                <ResourceDictionary Source="/Misc/DataRenderTemplates.xaml"/>
            </ResourceDictionary.MergedDictionaries>

        </ResourceDictionary>
    </UserControl.Resources>

While not strictly about application scope dictionaries, it seems to be the only decent reference to resource dictionaries in the WPF docs. That said, I was using the [UWP docs](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/resourcedictionary-and-xaml-resource-references) page about *ResourceDictionary and XAML resource references*, as it was literally the first result that came up when I googled "wpf resourcedictionary"

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 11633ee3-a647-aace-014c-5ed8bda58039
* Version Independent ID: a9544e6f-5248-203e-7554-ef1a6cf07116
* Content: [How to: Use an Application-Scope Resource Dictionary - WPF .NET Framework](https://docs.microsoft.com/en-us/dotnet/desktop/wpf/app-development/how-to-use-an-application-scope-resource-dictionary?view=netframeworkdesktop-4.8)
* Content Source: [dotnet-desktop-guide/framework/wpf/app-development/how-to-use-an-application-scope-resource-dictionary.md](https://github.com/dotnet/docs-desktop/blob/master/dotnet-desktop-guide/framework/wpf/app-development/how-to-use-an-application-scope-resource-dictionary.md)
* Product: **dotnet-framework**
* Technology: **dotnet-wpf**
* GitHub Login: @adegeo
* Microsoft Alias: **adegeo**

**Comment by adegeo** [comment]
Reading the comments on the stackoverflow question it seems the `DateTemplate` uses (and now that I think about it, it's probably the same for `Style`) the `DataType` as the key. Adding `x:Key` after a key has already been assigned doesn't change the key.

Actually, this is described in https://docs.microsoft.com/en-us/dotnet/desktop/wpf/advanced/xaml-resources-define?view=netframeworkdesktop-4.8#styles-datatemplates-and-implicit-keys which is linked to at the bottom of the article.

If I understood your question/problem/suggestion correctly, this is really the source of the problem, not the fact that there is an external file involved. Yes?

**Comment by butler1233** [comment]
That could be it. It's not something I've picked up on happening in the same file before, but that doesn't mean it doesn't happen, just that by chance I've not tried to set the key after something else already implicitly sets it. However, I do think it's something that would happen.

I know that when I usually do keyed resources, I set the `DataType`, not specifically so it will apply to everything, but usually just so the intellisense can give me something to work with. If it knows what properties I'm working with then life is nice and easy :)
